src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -400,16 +401,31 @@ export const handleGetPublicFormSampleSubmission: ControllerHandler<
 
     return res.status(statusCode).json({ message: errorMessage })
   }
+
   const form = formResult.value
 
   const publicForm = form.getPublicView() as PublicFormDto
 
   const formFields = publicForm.form_fields
   if (!formFields) {
-    throw new Error('unable to get form fields')
+    logger.error({
+      message: 'Form fields not found from form public view',
+      meta: logMeta,
+    })
+    return res.sendStatus(HttpStatusCode.InternalServerError)
```

Comment:
I feel that this should be a 4xx since the admin didn't provide any fields to sample. The actor for this error should be the admin instead of us (i.e., the request was invalid, rather than server error)

Reply:
Strike that.